package/MDAnalysis/converters/OpenMMParser.py

```diff
@@ -123,24 +136,58 @@ def _mda_topology_from_omm_topology(self, omm_topology):
         bond_orders = [b.order for b in omm_topology.bonds()]
         bond_types = [b.type for b in omm_topology.bonds()]
 
-        n_atoms = len(atomids)
-        n_residues = len(resids)
-        n_segments = len(segids)
-
         attrs = [
             Atomids(np.array(atomids, dtype=np.int32)),
             Atomnames(np.array(atomnames, dtype=object)),
-            Atomtypes(np.array(atomtypes, dtype=object)),
             Bonds(bonds, types=bond_types, order=bond_orders, guessed=False),
             ChainIDs(np.array(chainids, dtype=object)),
-            Elements(np.array(elements, dtype=object)),
-            Masses(np.array(masses, dtype=np.float32)),
             Resids(resids),
             Resnums(resnums),
             Resnames(resnames),
             Segids(segids),
         ]
 
+        validated_elements = []
+        masses = []
+        atomtypes = []
+        for a in omm_topology.atoms():
+            elem = a.element
+            if elem is not None:
+                if elem.symbol.capitalize() in SYMB2Z:
+                    validated_elements.append(elem.symbol.capitalize())
+                else:
+                    validated_elements.append('')
+                atomtypes.append(elem.symbol.capitalize())
+                masses.append(elem.mass._value)
+            else:
+                wmsg = (f"Unknown element {elem} found for some atoms. "
+                        f"These have been given an empty element record "
+                        f"with their atomtype set to 'X' "
+                        f"and their mass set to 0.0. "
+                        f"If needed they can be guessed using "
+                        f"MDAnalysis.topology.guessers.")
+                warnings.warn(wmsg)
```

Comment:
This warning will trigger for every invalid atom -- that sounds like it could get overwhelming. MDAnalysis already has a lot of warnings as it is. Could you please only warn once, if any element is not valid?

Reply:
I should've warned only once in the first place. The execution time for the tests reduced significantly after the redundant warnings were removed :)